### PR TITLE
[FSSDK-10579] replace Jest with Vitest

### PR DESCRIPTION
**Pull request by raju-opti**
## Summary
- Replace Jest test framework with Vitest

## Test plan
- Existing Jest tests has been updated to work with Vitest

## Issues
- FSSDK-10579

### CODE REVIEW

File: tests/pendingEventsDispatcher.spec.ts [hunk 1]
```diff
@@ -119,8 +122,8 @@ describe('PendingEventsDispatcher', () => {
       store,
       eventDispatcher: originalEventDispatcher,
     })
-    ;((getTimestamp as unknown) as jest.Mock).mockReturnValue(1)
-    ;((uuid as unknown) as jest.Mock).mockReturnValue('uuid')
+    ;((getTimestamp as unknown) as MockInstance).mockReturnValue(1)
+    ;((uuid as unknown) as MockInstance).mockReturnValue('uuid')
```
Comment (mikechu-optimizely) [inline tests/pendingEventsDispatcher.spec.ts:126]: NIT: Looks weir with the leading `;`  but for the sake of only Vitest edit....

File: tests/pendingEventsDispatcher.spec.ts [hunk 2]
```diff
@@ -30,29 +33,29 @@ import {
 import { EventDispatcher, EventV1Request } from '../lib/modules/event_processor/eventDispatcher'
 import { EventV1 } from '../lib/modules/event_processor/v1/buildEventV1'
 import { PendingEventsStore, LocalStorageStore } from '../lib/modules/event_processor/pendingEventsStore'
-import { uuid, getTimestamp } from '../lib/utils/fns'
+import { uuid, getTimestamp, objectValues } from '../lib/utils/fns'
```
Comment (mikechu-optimizely) [inline tests/pendingEventsDispatcher.spec.ts:36]: It looks like `objectValues` may not be being used. Maybe.

File: tests/reactNativeAsyncStorageCache.spec.ts [hunk 3]
```diff
@@ -14,7 +14,9 @@
  * limitations under the License.
  */
 
-/// <reference types="jest" />
+import { describe, beforeEach, beforeAll, it, vi, expect } from 'vitest';
+
+vi.mock('@react-native-async-storage/async-storage');
```
Comment (mikechu-optimizely) [inline tests/reactNativeAsyncStorageCache.spec.ts:19]: Is this a good time to remove the `__mocks__` directory too?

**Review by mikechu-optimizely — APPROVED** [review]
This is a lot of GREAT work @raju-opti. Kudos and thanks!

I have a few items below, but don't want to block your progress. LGTM.

File: tests/reactNativeAsyncStorageCache.spec.ts (hunk 3 above)
Comment (raju-opti) [reply tests/reactNativeAsyncStorageCache.spec.ts:19]: vi.mock is using the mock in __mock__ directory. it just does not do that automatically like jest, hence the explicit call

Comment (mikechu-optimizely) [reply tests/reactNativeAsyncStorageCache.spec.ts:19]: Ah. Good catch and learning for me. Thanks.

File: tests/pendingEventsDispatcher.spec.ts (hunk 1 above)
Comment (raju-opti) [reply tests/pendingEventsDispatcher.spec.ts:126]: code style in that file does not use semicolon at the end of statement, but in this particular case it is required. Let's keep it for now until we enforce consistent style across the whole project

Comment (mikechu-optimizely) [reply tests/pendingEventsDispatcher.spec.ts:126]: Arrrrgg ASI. Good eye.

File: tests/pendingEventsDispatcher.spec.ts (hunk 2 above)
Comment (raju-opti) [reply tests/pendingEventsDispatcher.spec.ts]: updated

File: tests/backoffController.spec.ts
[the remark this answers is not in the document]
Comment (raju-opti) [reply tests/backoffController.spec.ts:1]: updated

**Pull request merged**